Feature/site detail page

ericfennis opened the pull request:
## Icon Detail page!

- When you click now, your route will change and opens up the icon.
- Before it downloaded the icon, now you can choose if you want to Donwload or Copy.
- The key shortcuts still work when you click on an icon. CMD/ctrl + click and shift + click (maybe we need an hint for this somewhere on the website)
- I refactored and split up some code.
- @aelfric Your code works awesome, maybe some code of you is moved in to components.

![localhost_3000_icon_folder-plus](https://user-images.githubusercontent.com/11825403/96223340-09419e00-0f8e-11eb-9796-3531063b11ef.png)


**DEMO:**[deploy-preview-99--lucide.netlify.app]( https://deploy-preview-99--lucide.netlify.app/)

locness3 commented:
The website is missing the "context".

aelfric commented:
Is there something specific you want me to look at?

ericfennis commented:
@aelfric no sorry, Just want to inform you.

ericfennis commented:
@locness3 text is added again!

locness3 commented on site/src/components/Header.tsx:
```diff
@@ -25,8 +26,11 @@ const Header = ({ data }) => {
+        Simply beautiful open source icons, community-sourced
+      </Text>
+      <Text fontSize="lg" as="p" textAlign="center" mb="8">
+        An open-source icon library, a fork of Feather Icons. <br/>We're expanding the icon set as much as possible while keeping it nice-looking - <Link href="https://github.com/lucide-icons/lucide" isExternal>join us</Link>!
```
`Feather Icons` should be a link as well

delnyn commented:
The search bar is on the top of the details 
![image](https://user-images.githubusercontent.com/45879791/96404320-ebb54400-11da-11eb-9321-f83c445f6927.png)

delnyn commented:
Something is wrong at `/src/components/Layout.tsx:52:66`... "`Type error: No overload matches this call.`"

ericfennis commented:
Working on it!

ericfennis commented:
I Updated the site.
I also fixed the issue for #101 by updating it to a pre-release of Chakra UI where it is fixed, they used as wel on their own website.
One thing that I needed to change was the Icons in buttons, I used a beta release of our package. Which work fine I noticed. Later on I will update update the rest of the icons if I'm done with building it.

delnyn commented:
Great! But I tested the preview a bit and I think it's not detecting the system theme (after deleting local storage and reloading, it's not the system theme)

ericfennis commented:
Yeah you right, I forgot to test that. I will make an quick update for that tomorrow.

ericfennis commented:
It's working now, chakra fixed it!

delnyn commented:
Great 👏  
What is next?

ericfennis commented:
Merging!
